### ISSUE TITLE
The loss is "NAN" with Adam Optimizer.

### ISSUE THREAD

**Issue by stevenXS** [ultralytics/yolov5#6584]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

When I trained only several epochs using the Adam optimizer, losses became "nan".  And this is not the case with SGD. How to set the hyperparameter to avoid this situation? The hyperparameter settings are as follows:
- lr0: 0.001  
- lrf: 0.2  
- momentum: 0.937
- weight_decay: 0.0005  
- warmup_epochs: 3.0  
- warmup_momentum: 0.8 
- warmup_bias_lr: 0.1

**Comment by Hyrtsi** [comment]
Hi, we got the same results. I will post here if we find any solutions...

Meanwhile, `nan` with Adam seems to be "common":
- https://stackoverflow.com/questions/37232782/nan-loss-when-training-regression-network
- https://stackoverflow.com/questions/65643459/keras-nan-value-when-computing-the-loss
- https://stackoverflow.com/questions/40472499/issue-nan-with-adam-solver

**Comment by glenn-jocher** [comment]
@Hyrtsi thanks for sharing those links. The issue seems to be related to unstable training. Have you tried reducing the learning rate (`--lr`) and adjusting the weight decay (`--weight-decay`)? This combination has been effective for many users. Also, continue monitoring the discussion, as there might be more insights shared. Good luck!
